### PR TITLE
Migrate global_cache_tracker snapbox

### PR DESCRIPTION
**Pull request by dieterplex**
Part of #14039.

Some of these tests may require to assert specific removed file count or size that would be redact by current design. Need feedback to come out solution.

Temporary commit to pass the tests and illustrate what need to fix.

### CODE REVIEW

**Comment by epage** [comment]
> Temporary commit to pass the tests and illustrate what need to fix.

How about we skip those specific tests for now and focus on getting the rest of this in?

**Comment by dieterplex** [comment]
> > Temporary commit to pass the tests and illustrate what need to fix.
> 
> How about we skip those specific tests for now and focus on getting the rest of this in?

Added `#[allow(deprecated)]` to these tests.

File: tests/testsuite/global_cache_tracker.rs [hunk 1]
```diff
@@ -778,10 +778,24 @@ fn clean_gc_dry_run() {
...
+        let p = expected_files.replace(&root, "[ROOT]");
+        let p = if cfg!(windows) {
+            p.replace("\\", "/")
+        } else {
+            p
+        };
+        let p = re_crate.replace_all(&p, "-[HASH]");
+        let p = re_pack.replace_all(&p, "pack-[HASH]");
+        format!("{p}\n")
+    };
```
Comment (epage) [inline tests/testsuite/global_cache_tracker.rs:794]: Let's skip this one

Comment (dieterplex) [reply tests/testsuite/global_cache_tracker.rs]: done

Comment (dieterplex) [reply tests/testsuite/global_cache_tracker.rs]: Note for similar case https://github.com/rust-lang/cargo/pull/14096#discussion_r1646855403

File: crates/cargo-test-support/src/compare.rs [hunk 2]
```diff
@@ -196,7 +196,12 @@ fn add_regex_redactions(subs: &mut snapbox::Redactions) {
         regex!(r"\.cargo/target/(?<redacted>[0-9a-f]{2}/[0-9a-f]{14})"),
     )
     .unwrap();
-    subs.insert("[HASH]", regex!(r"/[a-z0-9\-_]+-(?<redacted>[0-9a-f]{16})"))
+    subs.insert(
+        "[HASH]",
+        regex!(r"/[a-z0-9\-_]+-(?<redacted>[0-9a-f]{16})\b"),
+    )
+    .unwrap();
+    subs.insert("[HASH]", regex!(r"/pack-(?<redacted>[0-9a-f]{40})\."))
```
Comment (epage) [inline crates/cargo-test-support/src/compare.rs:204]: This commit is last. Are tests passing in each commit and this commit is unused?

Comment (dieterplex) [reply crates/cargo-test-support/src/compare.rs:204]: Yes, it come with the skipped change to redact crate name and git pack filename correctly. Will dispose it.

**Pull request merged**